components/automate-cli/pkg/verifyserver/services/systemresourceservice/systemresourcesservice.go

```diff
@@ -0,0 +1,375 @@
+		resolutionMsg := fmt.Sprintf("CPU count should be >=%v", constants.MIN_CPU_COUNT)
+		checkResp = srs.GetChecksModel(false, CPU_COUNT_CHECK_TITLE, "", errorMsg, resolutionMsg)
+	}
+	return checkResp
+}
+
+func (srs *SystemResourcesService) GetCpuSpeedCheck(cpuInfoFile string) *models.Checks {
+	srs.logger.Debug("CPU speed check is running")
+	cpuSpeed, err := srs.GetOsAndFileSystemInfo.GetCPUSpeed(cpuInfoFile)
+	var resp *models.Checks
+
+	if err != nil {
+		srs.logger.Error("Error occured while getting cpu speed :", err)
+		ResolutionMsg := "Please run system on supported platform"
+		resp = srs.GetChecksModel(false, CPU_SPEED_CHECK_TITLE, "", err.Error(), ResolutionMsg)
+		return resp
+	}
+
+	srs.logger.Debug("CPU speed is : ", cpuSpeed)
+
+	if cpuSpeed >= constants.MIN_CPU_SPEED {
+		srs.logger.Debug("CPU speed is ", cpuSpeed)
+		successMsg := fmt.Sprintf("CPU speed should be >=%vGHz", constants.MIN_CPU_SPEED)
+		resp = srs.GetChecksModel(true, CPU_SPEED_CHECK_TITLE, successMsg, "", "")
+	} else {
+		srs.logger.Debug("CPU speed is ", cpuSpeed)
+		errorMsg := fmt.Sprintf("CPU speed is %vGHz", cpuSpeed)
+		resolutionMsg := fmt.Sprintf("CPU speed should be >=%vGHz", constants.MIN_CPU_SPEED)
+		resp = srs.GetChecksModel(false, CPU_SPEED_CHECK_TITLE, "", errorMsg, resolutionMsg)
+	}
+	return resp
+}
+
+func (srs *SystemResourcesService) GetMemorySizeCheck() *models.Checks {
+	srs.logger.Debug("Memory size check is running")
+	memoryInGB, err := srs.GetOsAndFileSystemInfo.GetMemory()
+	srs.logger.Debug("Current memory of system is :", memoryInGB)
+
+	var resp *models.Checks
+	if err != nil {
+		srs.logger.Error("Error occured while getting memory information : ", err.Error())
+		resolutionMsg := "Please run system on supported platform"
+		resp = srs.GetChecksModel(false, MEMORY_SIZE_CHECK_TITLE, "", err.Error(), resolutionMsg)
+		return resp
+	}
+
+	srs.logger.Debugf("Current memory of system is :%vGB", memoryInGB)
+	if memoryInGB >= constants.MIN_MEMORY {
+		successMsg := fmt.Sprintf("Memory should be >=%vGB", constants.MIN_MEMORY)
+		resp = srs.GetChecksModel(true, MEMORY_SIZE_CHECK_TITLE, successMsg, "", "")
+	} else {
+		errorMsg := fmt.Sprintf("Memory is %.2fGB", memoryInGB)
+		resolutionMsg := fmt.Sprintf("Memory should be >=%vGB", constants.MIN_MEMORY)
+		resp = srs.GetChecksModel(false, MEMORY_SIZE_CHECK_TITLE, "", errorMsg, resolutionMsg)
+	}
+	return resp
+}
+
+func (srs *SystemResourcesService) GetHabFreeSpaceCheckPreDeployment(nodeType string) (*models.Checks, error) {
+	srs.logger.Debug("Hab free space check is running for node_type : ", nodeType, " and deployment_state :", constants.PRE_DEPLOY)
+
+	currentFreeSpaceInGB, err := srs.GetFreeDiskSpaceOfGivenDir("/hab")
+	var resp *models.Checks
+	if err != nil {
+		srs.logger.Error("Unable to determine free space of /hab :", err)
+		resp = srs.GetChecksModel(false, fmt.Sprintf(FREE_SPACE_CHECK, "Hab"), "", err.Error(), "Please run system on supported platform")
```

Comment:
"Please run the system on the supported platform"
This is used in multiple places. Create a constant and use it to avoid code smell

Reply:
Done